crates/pico/src/memo_ref.rs

```diff
@@ -63,6 +127,83 @@ impl<T: 'static> Deref for MemoRef<T> {
             NodeKind::Derived(self.derived_node_id),
             revision.time_updated,
         );
-        value.downcast_ref::<T>().unwrap()
+        let mut any_ref: &dyn Any = value;
+        let len = self.projectors_len as usize;
+        for step in &self.projectors[..len] {
+            any_ref = (step)(any_ref);
+        }
+        any_ref
```

Comment:
hm. Can we optimize the case where `len == 0` and avoid the `any_ref = any_ref as dyn Any; any_ref.downcast_ref()`?